cpp/dolfinx/fem/FiniteElement.h

```diff
@@ -21,6 +21,15 @@ struct ufcx_finite_element;
 
 namespace dolfinx::fem
 {
+/// DOF transformation type
+enum class transformation_type
```

Comment:
`fem::transformation_type` is a bit generic and could mean quite a few things.

Maybe put it inside `FiniteElement` and call it something like `doftransform`? Could leave it in the `fem` namespace, in which case the name needs to be more specific. The former is probably better since it's used only in `FiniteElement`.

Reply:
@mscroggs ping